docker/importer/importer.py

```diff
@@ -452,8 +456,64 @@ def thread_init():
 
     logging.info("Finished processing bucket: %s", source_repo.name)
 
+  def _process_updates_rest(self, source_repo: osv.SourceRepository):
+    """Process updates from REST API."""
+    logging.info('Begin processing REST: %s', source_repo.name)
+    import_time_now = utcnow()
+    request = requests.head(source_repo.rest_api_url, timeout=_TIMEOUT_SECONDS)
+    if request.status_code != 200:
+      logging.error('Failed to fetch REST API: %s', request.status_code)
+      return
+    last_modified = datetime.datetime.strptime(request.headers['Last-Modified'],
+                                               _HTTP_LAST_MODIFIED_FORMAT)
+    # Check whether endpoint has been modified since last update
+    if last_modified < source_repo.last_update_date:
+      logging.info('No changes since last update.')
+      return
+    request = requests.get(source_repo.rest_api_url, timeout=_TIMEOUT_SECONDS)
+    # Get all vulnerabilities from the REST API. (CURL approach)
+    vulns = request.json()
+    # Create tasks for changed files.
+    for vuln in vulns:
+      import_failure_logs = []
+      last_modified = datetime.datetime.strptime(vuln['modified'],
+                                                 _OSV_MODIFIED_FORMAT)
+
+      if last_modified < source_repo.last_update_date:
+        continue
+      try:
+        vuln_location = source_repo.link + vuln['id'] + source_repo.extension
+        single_request = requests.get(vuln_location, timeout=_TIMEOUT_SECONDS)
+        _ = osv.parse_vulnerability_from_dict(single_request.json(),
+                                              source_repo.key_path,
+                                              self._strict_validation)
+        self._request_analysis_external(
+            source_repo, osv.sha256_bytes(vuln_location.encode()),
```

Comment:
The sha256 should be of the response body data from the single_request similar to bucket. 

The `git` version encodes the path because the commit hash already makes sure that the data doesn't change, but that doesn't apply for a REST endpoint.

Reply:
Makes sense! Just updated it now to use request.text to encode. :)